internal/ignition/ignition.go

```diff
@@ -1530,6 +1554,12 @@ func (g *installerGenerator) writeSingleHostFile(host *models.Host, baseFile str
 			return errors.Wrapf(errP, "Failed to parse machine cidr for node ip hint content")
 		}
 		setFileInIgnition(config, nodeIpHintFile, fmt.Sprintf("data:,KUBELET_NODEIP_HINT=%s", ip), false, 420, true)
+	} else if g.nodeIpAllocations != nil {
```

Comment:
This `else if` statement is a bit tricky, Are relaying on `g.nodeIpAllocations` to be nil in case of MNO with platform other than none?
It would be clearer if you check for the case you want to handle (none platform + MNO)

Reply:
In case of failure to allocate like https://github.com/openshift/assisted-service/pull/6257#discussion_r1606317112 the `g.nodeIpAllocations` will be nil.  So we will continue installation as usual.
So in general, in case we have a valid allocation, it should be used.
I will try to make the code clearer.

Reply:
Changed to rely also on cluster being multi-node-none-platform-cluster